src/test/java/uk/ac/sanger/sccp/stan/integrationtest/TestStainAndSubsequentMutations.java

```diff
@@ -90,23 +92,48 @@ public void testStainAndWorkProgressAndRecordResult() throws Exception {
         assertEquals(opId, result.getRefersToOpId());
         assertEquals(lw.getFirstSlot().getId(), result.getSlotId());
 
-        Integer permOpId = testPerm();
+        Integer permOpId = testPerm(lw);
         testVisiumAnalysis();
         Integer qcOpId = testVisiumQC(permOpId);
         testQueryVisiumQCResult(qcOpId);
     }
 
     // called by testStainAndWorkProgressAndRecordResult
-    private Integer testPerm() throws Exception {
+    private Integer testPerm(Labware lw) throws Exception {
+        var addControlOpType = entityCreator.createOpType("Add control", null);
+        Tissue tissue = entityCreator.getAny(tissueRepo);
+        Sample controlSample = entityCreator.createSample(tissue, 50);
+        Labware controlLabware = entityCreator.createLabware("STAN-C0", entityCreator.getTubeType(), controlSample);
+
         entityCreator.createOpType("Visium permabilisation", null, OperationTypeFlag.IN_PLACE);
         Object data = tester.post(tester.readGraphQL("perm.graphql"));
-        Integer opId = chainGet(data, "data", "recordPerm", "operations", 0, "id");
-        List<Measurement> measurements = measurementRepo.findAllByOperationIdIn(List.of(opId));
-        assertThat(measurements).hasSize(1);
-        Measurement meas = measurements.get(0);
-        assertEquals("permabilisation time", meas.getName());
-        assertEquals("120", meas.getValue());
-        return opId;
+        List<?> ops = chainGet(data, "data", "recordPerm", "operations");
+        assertThat(ops).hasSize(2);
+
+        Integer addControlId = chainGet(ops, 0, "id");
+        Integer permId = chainGet(ops, 1, "id");
+
+        Operation controlOp = opRepo.findById(addControlId).orElseThrow();
+        assertEquals(addControlOpType, controlOp.getOperationType());
+        assertThat(controlOp.getActions()).hasSize(1);
+        Action controlAction = controlOp.getActions().get(0);
+        assertEquals(controlLabware.getFirstSlot(), controlAction.getSource());
+        assertEquals(new Address(1,2), controlAction.getDestination().getAddress());
+        assertEquals(controlSample, controlAction.getSample());
+
+        List<Measurement> measurements = measurementRepo.findAllByOperationIdIn(List.of(permId));
+        assertThat(measurements).hasSize(2);
+        Measurement timeMeasurement = measurements.stream()
+                .filter(m -> m.getName().equalsIgnoreCase("permabilisation time"))
```

Comment:
spelling - permabilisation time instead of permeabilisation time

Reply:
Will need to be changed for x835, but can't be changed here because it will not work if it is spelled different from how it is in the database.